lib/redmine/default_data/loader.rb

```diff
@@ -212,21 +212,19 @@ def load(lang = nil)
             rejected  = Status.create!(name: l(:default_status_rejected), is_closed: true, is_default: false, position: 9)
             closed    = Status.create!(name: l(:default_status_closed), is_closed: true, is_default: false, position: 10)
 
-            # Workflow
-            statuses_for_task = [new, in_progress, on_hold, rejected, closed]
-            statuses_for_deliverable = [new, specified, in_progress, on_hold, rejected, closed]
-            statuses_for_none = [new, in_progress, rejected, closed]
-            statuses_for_milestone = [new, to_be_scheduled, scheduled, in_progress, on_hold, rejected, closed]
-            statuses_for_phase = statuses_for_milestone
-            statuses_for_bug = [new, confirmed, in_progress, tested, on_hold, rejected, closed]
-            statuses_for_feature = [new, specified, confirmed, in_progress, tested, on_hold, rejected, closed]
-            # Give each type its own workflow. Possible statuses are stored in one of the arrays above.
-            # Every status from the array gets a workflow to every other status from the array.
-            ['task', 'deliverable', 'none', 'milestone', 'phase', 'bug', 'feature'].each { |t|
-              (eval 'statuses_for_'.concat(t)).each { |os|
-                (eval 'statuses_for_'.concat(t)).each { |ns|
+            # Workflow - Each type has its own workflow
+            statuses = { task.name =>        [new, in_progress, on_hold, rejected, closed],
+                         deliverable.name => [new, specified, in_progress, on_hold, rejected, closed],
```

Comment:
You are using the types name as key here. The only use you make out of that is this:

````ruby
Type.where(:name => t).first.id
````

If you used e.g. `deliverable.id` here, you could simplify the other statement to

````ruby
Type.find(t).id # or find_by_id if you like
````
which could be simplified to:

````ruby
t
````